### PR TITLE
Foundation - Border radius fix for new discussion button

### PR DESCRIPTION
**Pull request by slafleche**
Border radius fix for new discussion button. It wasn't respecting the global variable.

Also fixed border radius issue with pagination previous button

Closes: https://github.com/vanilla/vanilla/issues/10246

### CODE REVIEW

File: applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts [hunk 1]
```diff
@@ -41,42 +41,64 @@ export const buttonCSS = () => {
...
+    mixinButton(".BoxButtons .ButtonGroup.Multi .Button.Primary", ButtonTypes.PRIMARY);
     mixinButton(".Section-Members .Group-RemoveMember", ButtonTypes.PRIMARY);
     mixinButton(".group-members-filter-box .Button.search", ButtonTypes.PRIMARY);
     mixinButton("#Form_Ban", ButtonTypes.PRIMARY);
     mixinButton(".Popup #UserBadgeForm button", ButtonTypes.PRIMARY);
     mixinButton(".Button.Handle", ButtonTypes.PRIMARY);
     mixinButton("div.Popup .Body .Button.Primary", ButtonTypes.PRIMARY);
     mixinButton(".ButtonGroup.Multi .Button.Handle", ButtonTypes.PRIMARY);
-    mixinButton(".ButtonGroup.Multi .Button.Handle .Sprite.SpDropdownHandle", ButtonTypes.PRIMARY);
+    // mixinButton(".ButtonGroup.Multi .Button.Handle .Sprite.SpDropdownHandle", ButtonTypes.PRIMARY);
```
Comment (alex-mtl) [inline applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts]: Usually we don't push commented code. Is it intentional here?

File: applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts [hunk 2]
```diff
@@ -41,42 +41,64 @@ export const buttonCSS = () => {
...
         ...borders({
-            ...globalVars.borderType.formElements.buttons,
             color: rgba(0, 0, 0, 0),
         }),
     });
 
+    cssOut(`.ButtonGroup.Multi.NewDiscussion .Button.Handle.Handle`, {
+        top: unit(0),
+        right: unit(formElementVars.border.width),
+        minWidth: importantUnit(handleSize),
```
Comment (alex-mtl) [inline applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts]: I can predict `veryImportantUnit` coming soon? (just a joke)
But naming convention is not clear. what is important and what is not?
Especially when we talk about some utils function like `unit`.
What function should I use next time I need to get unit?

Comment (charrondev) [reply applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts]: @alex-mtl This maps to CSS `!important`. https://css-tricks.com/when-using-important-is-the-right-choice/

File: applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts (hunk 1 above)
Comment (charrondev) [reply applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts]: Yeah this should go.
```suggestion
```

File: packages/vanilla-theme-core/scss/components/_forms.scss [hunk 3]
```diff
@@ -317,8 +317,10 @@ div.token-input-dropdown {
     > .Button:first-child {
         width: 100%;
         text-align: center;
-        border-top-right-radius: $formButton_borderRadius;
-        border-bottom-right-radius: $formButton_borderRadius;
+        @if($staticVariables) {
```
Comment (charrondev) [inline packages/vanilla-theme-core/scss/components/_forms.scss]:
```suggestion
        @if ($staticVariables) {
```

File: applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts [hunk 4]
```diff
@@ -41,42 +41,64 @@ export const buttonCSS = () => {
...
     mixinButton(".group-members-filter-box .Button.search", ButtonTypes.PRIMARY);
     mixinButton("#Form_Ban", ButtonTypes.PRIMARY);
     mixinButton(".Popup #UserBadgeForm button", ButtonTypes.PRIMARY);
     mixinButton(".Button.Handle", ButtonTypes.PRIMARY);
     mixinButton("div.Popup .Body .Button.Primary", ButtonTypes.PRIMARY);
     mixinButton(".ButtonGroup.Multi .Button.Handle", ButtonTypes.PRIMARY);
-    mixinButton(".ButtonGroup.Multi .Button.Handle .Sprite.SpDropdownHandle", ButtonTypes.PRIMARY);
+    // mixinButton(".ButtonGroup.Multi .Button.Handle .Sprite.SpDropdownHandle", ButtonTypes.PRIMARY);
+
+    const buttonBorderRadius = parseInt(formElementVars.border.radius.toString(), 10);
```
Comment (charrondev) [inline applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts]: Shouldn't this be the `buttonGlobalVars().border.radius`?

File: applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts (hunk 2 above)
Comment (alex-mtl) [reply applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts]: Got it.

File: applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts (hunk 4 above)
Comment (slafleche) [reply applications/dashboard/src/scripts/compatibilityStyles/buttonStylesCompat.ts]: You're right, it's not the right value. however, it should be globalVars.borderType.formElements.buttons.

This is the hierarchy:
```
    const border = makeThemeVars("border", {
        color: mixBgAndFg(0.15),
        colorHover: mixBgAndFg(0.2),
        width: 1,
        style: "solid",
        radius: 6, // Global default <=== Default for all
    });

    const borderType = makeThemeVars("borderType", {
        formElements: {
            default: border, // <=== Default for form elements (like inputs)
            buttons: border, // <=== Defaults for buttons
        },
        modals: border, // <=== Defaults for modals
        dropDowns: {
            content: border, // <=== Defaults for dropdown content
        },
    });
```

**Pull request merged**